### DISCUSSION TITLE
Genmon working with Apple HomeKit via HomeBridge

### DISCUSSION THREAD

**Discussion by acesarewld** [jgyates/genmon#1108] [Ideas]
Hi All,
I wanted to tell the group that I got my Genmon system working with Apple HomeKit via a HomeBridge plugin today. Here's what I did...

The plugin I installed is called [homebridge-cmdswitch2 ](https://github.com/luisiam/homebridge-cmdswitch2) and I used a few Genmon API endpoints to get things working.

Examples of endpoints I used...

Exercise Generator: http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=startexercise
Start Generator: http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=start
Start + Transfer: http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=starttransfer
Stop: http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=stop

I setup the Homebridge plugin like this: 
{
    "platform": "cmdSwitch2",
    "name": "CMD Switch",
    "synchronous": true,
    "switches": [
        {
            "name": "Exercise Generator",
            "on_cmd": "curl -X GET 'http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=startexercise'",
            "off_cmd": "curl -X GET 'http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=stop'"
        },
        {
            "name": "Generator",
            "on_cmd": "curl -X GET 'http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=start'",
            "off_cmd": "curl -X GET 'http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=stop'"
        },
        {
            "name": "Transfer to Generator",
            "on_cmd": "curl -X GET 'http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=starttransfer'",
            "off_cmd": "curl -X GET 'http://<YOUR_GENMON_SYSTEM>/cmd/setremote?setremote=stop'"
        }
    ]
}

It works great, although it only shows the ON / OFF status based on you flipping the switch — not feedback from GenMon.
I was thinking I can write a query API that uses the engine state from the following endpoint: 

http://<YOUR_GENMON_SYSTEM>/cmd/status_json

Anyway, it's up and working, I've got automations setup to turn on my Generac and transfer when my Tesla Powerwalls hit 15% so there will never be any time without power.

Let me know if you have any thoughts or questions! 
-G²

**Comment by jgyates** [comment]
Great info, thanks for the post!
